pgmax/groups/variables.py

```diff
@@ -106,29 +176,57 @@ class VariableDict(groups.VariableGroup):
 
     Args:
         num_states: The size of the variables in this variable group
-        variable_names: A tuple of all names of the variables in this variable group
-
+        variable_names: A tuple of all names of the variables in this variable group.
+            Note that we overwrite variable_names to add the hash of the VariableDict
     """
 
-    num_states: int
     variable_names: Tuple[Any, ...]
+    num_states: int
```

Comment:
Any reason not to support variable number of states? Should be simple and the same as `NDVariableArray`